### PR TITLE
Fixed trivy vulnerabilities. Update go modules, golang to 1.23 

### PR DESCRIPTION
**Pull request by deepak07dbz**
Changed repo target from docker to quay and upgraded golang version to `1.23` as mentioned in trivy vulnerabilities.

`$ k get po -A | grep whereabouts`
`luigi-system             whereabouts-lxm9z                           1/1     Running   0              46m`

`$ k describe -n luigi-system po/whereabouts-lxm9z | grep -i image`
`    Image:         quay.io/platform9/whereabouts:v0.6-pmk-3427514`
`    Image ID:     quay.ioplatform9whereabouts@sha256:2b0de78f3b9f6b43ca0e62afbb3b4b0050358bf7631b06bd4c4acf52d805820a`
`  Normal   Pulling           45m   kubelet            Pulling image "quay.io/platform9/whereabouts:v0.6-pmk-3427514"`
`  Normal   Pulled            45m   kubelet            Successfully pulled image "quay.io/platform9/whereabouts:v0.6-pmk-3427514" in 5.931958779s (5.931966272s including waiting)`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.20
+FROM golang:1.23.1
```
Comment (cruizen) [inline Dockerfile]:
```suggestion
FROM golang:1.23
```

**Comment by cruizen** [comment]
@deepak07dbz The tests from github actions were failing. Made some changes to also run `hack/update-codegen.sh`

**Review by cruizen — APPROVED** [review]

**Comment by cruizen** [comment]
@deepak07dbz Please add the path to image generated on quay.io and run `trivy image` against this image tag. 

Also. update the image version in airctl in https://github.com/platform9/airctl/blob/atherton/conf/pmk_cluster_img_list.json with a follow-up PR

**Review by mithilarun — APPROVED** [review]

**Comment by cruizen** [comment]
Please go ahead and `squash and merge`. Make a separate PR in airctl 5.11 branch to update the list of images. Raise a pull request with similar testing notes.

**Pull request merged**